### PR TITLE
Raise an Award and Award raised by me 

### PR DESCRIPTION
**Pull request by Amruta1854**
Raise an Award and Award raised by me are integrated and working.

### CODE REVIEW

File: Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java [hunk 1]
```diff
@@ -1,14 +1,133 @@
...
+	}
+
+	@Override
+	public AwardRequest raiseAward(AwardRequest awardRequest) {
+		ErrpUser user = userService.getUserById(getLoggedInUser().getEmployeeId());
+		ErrpUser userSupervisor = userService.getUserById(awardRequest.getAwardee().getEmployeeId());
+		awardRequest.setCreatedDate(new Date());
+		awardRequest.setUpdatedDate(new Date());
+		awardRequest.setAwardRaisedBy(user);
+		awardRequest.setAwardRequestStatus(0);
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java:74]: What does 0 status mean, externalize and use an Enum

File: Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java [hunk 2]
```diff
@@ -1,14 +1,133 @@
...
+	public AwardRequest updateRequest(AwardRequest awardRequest) {
+		System.out.println(awardRequest.getAward());
+		if (validateRequest(awardRequest.getRequestId())) {
+			Optional<AwardRequest> optionalAwardRequest = this.awardRequestRepository
+					.findById(awardRequest.getRequestId());
+			AwardRequest updateRequest = optionalAwardRequest.get();
+			updateRequest.setUpdatedDate(new Date());
+			updateRequest.setRemarks(awardRequest.getRemarks());
+			updateRequest.setAward(awardRequest.getAward());
+			System.out.println(awardRequest.getAward());
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java:110]: Remove sysout

File: Server/src/main/java/com/abs/errp/awardtype/AwardTypePublicController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.abs.errp.entity.AwardType;
+
+@RestController
+@RequestMapping("/awardTypesPublic")
+public class AwardTypePublicController {
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/awardtype/AwardTypePublicController.java:13]: What is Public? what does this do? Please find appropriate name

File: Server/src/main/java/com/abs/errp/entity/AwardRequest.java [hunk 4]
```diff
@@ -1,32 +1,42 @@
...
-	private int awardRaisedBy;
+	@OneToOne(cascade = CascadeType.DETACH)
+	@JoinColumn(name = "AWARD_RAISED_BY_ID", referencedColumnName = "EMPLOYEE_ID")
+	private ErrpUser awardRaisedBy;
 
-	@Column(name = "SUBMITTED_TO_ID")
-	private int submittedToId;
+	@OneToOne(cascade = CascadeType.DETACH)
+	@JoinColumn(name = "SUBMITTED_TO_ID", referencedColumnName = "EMPLOYEE_ID")
+	private ErrpUser submittedToId;
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/entity/AwardRequest.java:39]: Naming convention - if it is of type ErrpUser should not have the name submittedToId - only submittedTo will indicate

File: UI/src/app/award/award.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+// import { TestBed } from '@angular/core/testing';
+
+// import { AwardService } from './awardrequest.service';
+
+// describe('AwardService', () => {
```
Comment (rbaidABS) [inline UI/src/app/award/award.service.spec.ts:5]: why is this test file commented?

**Review by rbaidABS — CHANGES_REQUESTED** [review]

File: Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java [hunk 6]
```diff
@@ -1,14 +1,129 @@
...
+	}
+
+	@Override
+	public AwardRequest raiseAward(AwardRequest awardRequest) {
+		ErrpUser user = userService.getUserById(getLoggedInUser().getEmployeeId());
+		ErrpUser userSupervisor = userService.getUserById(awardRequest.getAwardee().getEmployeeId());
+		awardRequest.setCreatedDate(new Date());
+		awardRequest.setUpdatedDate(new Date());
+		awardRequest.setAwardRaisedBy(user);
+		awardRequest.setAwardRequestStatus(AwardRequestStatus.PENDING.ordinal());
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/awardrequest/AwardRequestServiceImpl.java:72]: This is a tight coupling between the position of the status defined in the enum. What will happen if the sequence in the enum is changed or If I add a new status right at the start in the enum? Please create enum with id , status mapping.

File: Server/src/main/java/com/abs/errp/awardrequest/AwardRequestStatus.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package com.abs.errp.awardrequest;
+
+public enum AwardRequestStatus {
```
Comment (rbaidABS) [inline Server/src/main/java/com/abs/errp/awardrequest/AwardRequestStatus.java:3]: There should be explicit mapping between Id and description of status

**Review by rbaidABS — CHANGES_REQUESTED** [review]

**Review by rehannawab — CHANGES_REQUESTED** [review]

**Review by rehannawab — APPROVED** [review]

**Review by rehannawab — APPROVED** [review]

**Pull request merged**